### ISSUE TITLE
Fix OS/2 (NE 16bit) argument passing

### ISSUE THREAD

**Issue by roytam1** [Baron-von-Riedesel/HX#2]
There is some effort that improves HX:

Fix OS/2 (NE 16bit) argument passing 
https://github.com/Ringdingcoder/hx-for-opus/commit/704231af4028d9c42c608b3bff9d138395b06458

Another OS/2 argument passing fix (DosExecPgm) 
https://github.com/Ringdingcoder/hx-for-opus/commit/c1a72c5338b217a52c0fab66ae88c27e2fbcbb4c

**Comment by Baron-von-Riedesel** [comment]
How can I test the fix(es)?

**Comment by Ringdingcoder** [comment]
Invoke one OS/2 16bit NE from another one and pass some command line args.

The second patch (to DOSXXX) is pretty obvious because it destroys the ax register which is later used by scasb.

The other one is in the DPMI loader's interrupt hook. Your version copies the command line to offset 80h in the PSP, but that’s not where the OS/2 startup code expects it. That’s why I copy it over into the environment segment. Without this, the invoked command would see the same command line as its caller.

**Comment by Ringdingcoder** [comment]
From looking at this now, it looks like the loader patch copies only the first argument. I'm not sure right now, and I cannot investigate. The "fix" is certainly not entirely correct, but it made the use case work for which I needed it (running the Opus OS/2 compiler).

**Comment by Ringdingcoder** [comment]
No, it copies the entire line. The additional cmp loop is for skipping over the first argument (the command name), and it scans the previous environment segment's contents, i.e. the one used to invoke the current program. The code should be mostly fine, unless the new line is too long for the segment, in which case it will likely crash.

The good way would be to allocate a new environment segment and accumulate everything there. I have seen an #ifdef to this effect, but have not investigated further.
